src/EditorFeatures/CSharpTest/EmbeddedLanguages/RegularExpressions/CSharpRegexParserTests_RealWorld.cs

```diff
@@ -0,0 +1,45 @@
+                RegexOptions.None, conversionFailureOk: false);
+            Assert.NotNull(tree);
+            Assert.Empty(tree.Diagnostics);
+        }
+
+        [Fact]
+        public void TestRealWorldCases()
+        {
+            using var stream = this.GetType().Assembly.GetManifestResourceStream("Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.EmbeddedLanguages.RegularExpressions.Regex_RealWorldPatterns.json");
+            using var streamReader = new StreamReader(stream);
+            using var textReader = new JsonTextReader(streamReader);
+
+            foreach (var obj in JArray.Load(textReader))
+            {
+                var options = (RegexOptions)obj.Value<int>("Options");
+                var pattern = obj.Value<string>("Pattern")!.Replace(@"""", @"""""");
+                var (_, tree, _) = JustParseTree($@"@""{pattern}""", options, conversionFailureOk: false);
+                Assert.NotNull(tree);
```

Comment:
Consider changing this to use `[MemberData(..)]` so that each input is a separate test, and failures can be more easily identified.

Unless of course the potential failures all-or-nothing, but at the moment this test will fail at the first of 19000 inputs with any kind of failure, and with no messages about which input is the problematic one, and without running any subsequent tests.

Reply:
let me try :)

Reply:
done!